api/src/services/relations.ts

```diff
@@ -78,20 +94,26 @@ export class RelationsService {
 	}
 
 	async readOne(collection: string, field: string): Promise<Relation> {
-		if (this.accountability && this.accountability.admin !== true) {
-			if (this.hasReadAccess === false) {
-				throw new ForbiddenError();
-			}
-
-			const permissions = this.accountability.permissions?.find((permission) => {
-				return permission.action === 'read' && permission.collection === collection;
-			});
+		if (this.accountability) {
+			await validateAccess(
+				{
+					accountability: this.accountability,
+					action: 'read',
+					collection: 'directus_relations',
+				},
+				{
+					schema: this.schema,
+					knex: this.knex,
+				},
+			);
 
-			if (!permissions || !permissions.fields) throw new ForbiddenError();
+			const allowedFields = await fetchAllowedFields(
+				{ collection, action: 'read', accountability: this.accountability },
+				{ schema: this.schema, knex: this.knex },
+			);
 
-			if (permissions.fields.includes('*') === false) {
-				const allowedFields = permissions.fields;
-				if (allowedFields.includes(field) === false) throw new ForbiddenError();
+			if (allowedFields.includes(field) === false) {
```

Comment:
This is missing the check for `'*'`. Was that done on purpose? As far as I can see `fetchAllowedFields` does not expand `'*'` to all available fields of a collection, so explicitly checking for `'*'` seems necessary?